lib/include/tree_sitter/api.h

```diff
@@ -1166,6 +1166,26 @@ uint32_t ts_language_symbol_count(const TSLanguage *self);
 */
 uint32_t ts_language_state_count(const TSLanguage *self);
 
+/**
+ * Get the number of supertype symbols in the language.
+*/
+uint32_t ts_language_supertype_count(const TSLanguage *self);
+
+/**
+ * Get a list of all supertype symbols for the language.
+*/
+const TSSymbol *ts_language_supertypes(const TSLanguage *self);
```

Comment:
Instead of exposing a function for just getting the # of supertypes (which I question the usefulness of), why not instead have the caller pass in a pointer to a `uint32_t`, called `length`, which is written to in the function body. This will let the user know how many supertypes there are, see `ts_tree_included_ranges` for an example of what I mean. The main reason being this will set the precedence for us to add `_count` functions for other node types, which we don't want to add.